### PR TITLE
Implement IMG ops, Fix autogenerated docs, Update symbolic shape inference

### PR DESCRIPTION
**Pull request by orausch**

### CODE REVIEW

File: daceml/onnx/op_implementations/img_op_implementations.py [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+                                          y=dace.Memlet("Y[{}]".format(
+                                              ", ".join("i{}".format(i)
+                                                        for i, _ in enumerate(Y.shape))))
+                                      ),
+                                      external_edges=True)
+        # yapf: enable
+
+        # the outer map loops over every entry in the output array
+        outer_me, outer_mx = new_state.add_map(
+            'outer_conv_map',
```
Comment (tbennun) [inline daceml/onnx/op_implementations/img_op_implementations.py:100]: not exactly conv (also below)

File: daceml/onnx/op_implementations/img_op_implementations.py [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+        read_X = new_state.add_read("X")
+        inner_image_memlet = propagation.propagate_memlet(
+            new_state, image_memlet, inner_me, False)
+        outer_image_memlet = propagation.propagate_memlet(
+            new_state, inner_image_memlet, outer_me, False)
+        new_state.add_edge(outer_me, None, inner_me, None, inner_image_memlet)
+        new_state.add_edge(read_X, None, outer_me, None, outer_image_memlet)
+
+        # hook up outputs
+        output_memlet = dace.Memlet("Y[b, c, out_x, out_y]",
```
Comment (tbennun) [inline daceml/onnx/op_implementations/img_op_implementations.py:138]: hopefully this should just work (I've seen cases where it doesn't validate with cudnn because of "what happens if two values are equal" in the backprop)

File: daceml/onnx/op_implementations/img_op_implementations.py [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+        if len(X.shape) != 4 or len(W.shape) != 4:
+            return False
+
+        if node.group != 1:
+            return False
+
+        if num_channels != W.shape[1]:
+            return False
+
+        if node.dilations is not None and (not all(d == 1
```
Comment (tbennun) [inline daceml/onnx/op_implementations/img_op_implementations.py:187]: do we need those for any of the current use cases?

File: daceml/transformation/constant_folding.py [hunk 4]
```diff
@@ -140,6 +140,8 @@ def apply(self, sdfg: dace.SDFG):
             sub_state = sub_sdfg.add_state()
 
             node_copy = copy.deepcopy(node)
+            # always use ORT for constant folding
+            node_copy.implementation = "onnxruntime"
```
Comment (tbennun) [inline daceml/transformation/constant_folding.py:144]: not a big fan if it requires us to have ort installed for any daceml operation

File: tests/pytorch/test_lenet.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+        x = x.view(-1, 16 * 6 * 6)
+        x = F.relu(self.fc1(x))
+        x = F.relu(self.fc2(x))
+        x = self.fc3(x)
+        x = F.log_softmax(x, dim=1)
+        return x
+
+
+@pytest.mark.parametrize("conv_impl", ["pure", "im2col"])
```
Comment (tbennun) [inline tests/pytorch/test_lenet.py:35]: I have not seen the im2col implementation anywhere. Where is it defined?

File: daceml/onnx/op_implementations/img_op_implementations.py (hunk 3 above)
Comment (orausch) [reply daceml/onnx/op_implementations/img_op_implementations.py:187]: We'll add them as we need them

File: daceml/transformation/constant_folding.py (hunk 4 above)
Comment (orausch) [reply daceml/transformation/constant_folding.py:144]: Unfortunately the dace implementations are still not stable enough to work for all cases (especially in edge cases with scalars, which is where constant folding is usually applied)

File: tests/pytorch/test_lenet.py (hunk 5 above)
Comment (orausch) [reply tests/pytorch/test_lenet.py:35]: I removed it because it doesn't work #49.

File: daceml/onnx/onnx_importer.py [hunk 6]
```diff
@@ -385,13 +385,25 @@ def __call__(self, *args,
 
         inputs, params, symbols, outputs = self._call_args(args=args,
                                                            kwargs=kwargs)
+        transient_kwargs = {}
+        if self.save_transients is not None:
+            for node, parent in self.sdfg.all_nodes_recursive():
+                if isinstance(node, nodes.AccessNode):
+                    desc = self.sdfg.arrays[node.data]
+                    if desc.transient and dace.can_access(
```
[the remark this answers is not in the document]
Comment (orausch) [reply daceml/onnx/onnx_importer.py]: Removed the condition and added a test.

File: tests/pytorch/test_lenet.py (hunk 5 above)
Comment (tbennun) [reply tests/pytorch/test_lenet.py:35]: shouldn't we remove the parameterization too?

Comment (orausch) [reply tests/pytorch/test_lenet.py:35]: I plan to re-add it, so I guess we can keep it until that happens

File: daceml/onnx/op_implementations/img_op_implementations.py (hunk 1 above)
Comment (tbennun) [reply daceml/onnx/op_implementations/img_op_implementations.py:100]: @orausch don't you want to change it to `outer_pool_map` etc.?

Comment (orausch) [reply daceml/onnx/op_implementations/img_op_implementations.py]: oops, changed

**Review by tbennun — APPROVED** [review]
